### PR TITLE
Refactor and test db commands

### PR DESCRIPTION
**Pull request by calonso**

### CODE REVIEW

File: database/database.go [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+	req, err := http.NewRequest("GET", fmt.Sprintf("http://%s:5986/_dbs/%s", ahr.GetServer(), db.name), nil)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	ahr.RunRequest(req, &db.config)
+}
+
+func (db *Database) Replicate(shard, replica string, ahr *http_utils.AuthenticatedHttpRequester) error {
+	replica_node := fmt.Sprintf("couchdb@%s", replica)
```
Comment (sayden) [inline database/database.go]: Go encourages CameCased variables

File: database/database.go [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+
+	if array_utils.Contains(db.config.ByNode[replica_node], shard) {
```
Comment (sayden) [inline database/database.go]: Same here. Go encourages CamelCased variables. `sliceUtils` should be more correct

**Review by sayden — COMMENTED** [review]
Generally OK++

Check port 5986 where mentioned

File: database/database.go [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+		delete(db.config.ByNode, replica)
+	}
+	// TODO add an entry to the changes section.
+
+	b, err := json.Marshal(db.config)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	req, err := http.NewRequest("PUT", fmt.Sprintf("http://%s:5986/_dbs/%s", ahr.GetServer(), db.name), bytes.NewBuffer(b))
```
Comment (sayden) [inline database/database.go:119]: Are you sure it's port 5986 instead of 5984?

Comment (calonso) [reply database/database.go:119]: Yep, Is the backdoor port!! :)

**Review by sayden — APPROVED** [review]
LGTM

**Pull request merged**